src/position_control.hpp

```diff
@@ -28,6 +29,133 @@ static THD_FUNCTION(PositionControlThread, arg) {
     }
 }
 
+/**
+ * All the conversions between the different coordinates are given below, we can move them to whatever file we want
+ * This needs to define global variable for the motor angles, leg angles, and foot coordinates so the functions are void
+ * We also have to define the upper and lower leg lengths L1 and L2 as global vars
+ * Here the motor angles are assumed to be in radians with a 0 position in the x-direction along the body and positive angles CW
+ * Make a MotorAngles struct with th1 and th2 values
+ * The x,y cartesian coordinates have the x-direction pointed where the robot is heading and y-direction is positive towards the ground
+ * Make struct with two floats for LegCartesian and give an x,y values
+ * The leg length (L) is the virtual leg length and is always positive, the leg angle, gamma, is zero in the y-direction and positive when going CW
+ * Make a struct with L,gamma values called LegParams
+ */
+float L1 = 0.15 // upper leg length (m) what is it actually?
+float L2 = 0.3 // lower leg length (m)
+
+/**
+ * TODO: Need to change the th1, th2 values as defined above to the encoder counts to set for the position controller. Encoder counts per rev is 2000
+ * Make this an if else and handle different orientations, offsets based on which motor is given in motorIndex
+ */
+int MotorAngleToEncoderCounts(theta, thetaIndex, motorIndex) {
+  // have an if else based on if thetaIndex is 0 or 1 for either theta so just return 1 value and avoid matrix
+}
+
+/**
+ * TODO: Need to change the encoder counts enc1, enc2 values to the motor angles as defined above
+ * Make this an if else and handle different orientations, offsets based on which motor is given in motorIndex
+ */
+float EncoderCountsToMotorAngle(enc, encIndex, motorIndex) {
+}
+
+/**
+ * Converts the two motor angles (in radians) for a leg to the cartesian coordinates
+ */
+void MotorAngleToCartesian(th1, th2) {
+  delta = (th2+th1+PI)/2.0 - acos(L1/L2*sin((th2-th1)/2.0));
+  x = L1*cos(th1) + L2*cos(delta);
+  y = L1*sin(th1) + L2*sin(delta);
+}
+
+/**
+ * Converts the two motor angles (in radians) for a leg into the motor angles th1, th2 (in rads)
+ */
+void LegParamsToMotorAngles(L, gamma) {
+  beta = 2.0*acos((L1^2.0+L^2.0-L2^2.0)/(2.0*L1*L));
+  th1 = (PI/2.0 + gamma - beta/2.0);
+  th2 = (PI/2.0 + gamma + beta/2.0);
+}
+
+/**
+ * Converts the leg params L, gamma to cartesian coordinates x, y (in m)
+ */
+void LegParamsToCartesian(L, gamma) {
+  x = -L*sin(gamma);
+  y = L*cos(gamma);
+}
+
+/**
+ * Converts the cartesian coords x, y (m) to leg params L (m), gamma (rad)
+ */
+void CartesianToLegParams(x, y) {
```

Comment:
We've been treating y as positive downward but let's discuss our sign conventions on Sunday!

Reply:
Perfect I was also using y positive downwards, but yeah we can fix my angle conventions.